getent: Handle non-empty split string

vvinayakkk opened the pull request:
Fixes #85720: Handle empty split string in getent module gracefully. Adds validation and unit test. See issue for details.

vvinayakkk committed:
getent: fail gracefully on empty split (fixes #85720)

Signed-off-by: Vinayak Bhatia <ntpjc2vinayak@gmail.com>
Co-authored-by: Abhijeet Kasurde <Akasurde@redhat.com>

bcoca commented on lib/ansible/modules/getent.py:
```diff
@@ -138,6 +139,14 @@ def main():
 
+    # Validate split early: Python str.split("") raises ValueError("empty separator")
+    # Provide a clean module error instead of an unhandled traceback.
+    if split == "":
```
instead of this, i would change to `if not split` in line 151, which would cover more 'invalid' inputs, mabye adding a warning instead of an error `if split is not None:  warning='Ignoring invalid split option ({split!r}), using autodetection/'`

vvinayakkk commented:
Update: Implemented reviewer feedback (treat falsy/invalid split as autodetect and emit a warning), fixed module docs (validate-modules), updated unit test to assert success and parsing, and adjusted changelog. Removed a brittle assertion on warning text. CI is now green across sanity and units. Thanks!

Akasurde commented on lib/ansible/modules/getent.py:
```diff
@@ -34,8 +34,10 @@
+                                                - Character used to split the database values into lists/arrays such as ':' or a tab character,
+                                                        otherwise it will try to pick one depending on the database.
+                                                - Falsy/invalid values (for example an empty string) will be ignored with a warning and
+                                                    autodetection will be used.
```
```suggestion
            - Character used to split the database values into lists/arrays such as ':' or a tab character,
              otherwise it will try to pick one depending on the database.
            - Falsy/invalid values (for example an empty string) will be ignored with a warning and
              autodetection will be used.
```

Akasurde commented on lib/ansible/modules/getent.py:
```diff
@@ -148,8 +158,7 @@ def main():
 
-    if split is None and database in colon:
-        split = ':'
+    # split already normalized above; nothing else to do here
```
```suggestion

```

Akasurde replied:
the comment seems redundant here.

Akasurde commented on test/units/modules/test_getent.py:
```diff
@@ -0,0 +1,34 @@
+from __future__ import annotations
```
I would write an integration test [here](https://github.com/ansible/ansible/blob/devel/test/integration/targets/getent/tasks/main.yml)

Akasurde commented:
@vvinayakkk Are you still working on this? Thanks.

Akasurde committed:
Review request

Signed-off-by: Abhijeet Kasurde <Akasurde@redhat.com>

Akasurde committed:
Make CI green I

Signed-off-by: Abhijeet Kasurde <Akasurde@redhat.com>
